### ISSUE TITLE
Type of @property that returns a list is not correctly inferred in Generic class

### ISSUE THREAD

**Issue by ldorigo** [microsoft/pylance-release#2841]
Issue Type: <b>Bug</b>

Hi, 

This is most likely an issue with Pyright but I'm posting the issue here to get VSCode's diagnostics information.

Consider this snippet:

```python
from typing import List, TypeVar, Generic

SomeType = TypeVar("SomeType")

class A(Generic[SomeType]):
    a: SomeType

    @property
    def a_list(self) -> List[SomeType]:
        return [self.a] * 5

    def __init__(self, a : int):
        self.a = a
        
    def test(self):
        fails_typecheck = [el for el in self.a_list]
        

```

The line `fails_typecheck` gives the following error(s):

```
Argument type is partially unknown
  Argument corresponds to parameter "__obj" in function "len"
  Argument type is "list[Unknown]" - [PylancereportUnknownArgumentType]
------
Type of "el" is unknown - [PylancereportUnknownVariableType]
```

I believe the type checker should be able to infer that `el` is of type `SomeType`? I tested the same code with mypy, and it doesn't show any errors.

Extension version: 2022.5.2
VS Code version: Code 1.67.2 (c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5, 2022-05-17T18:23:40.286Z)
OS version: Linux x64 5.15.37-1-lts
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 3700X 8-Core Processor (16 x 4227)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 1|
|Memory (System)|31.36GB (3.84GB free)|
|Process Argv|--crash-reporter-id 66ea3ec9-05eb-4edd-8d35-34bbe6c1aba1|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|plasma-i3|
|XDG_CURRENT_DESKTOP|KDE|
|XDG_SESSION_DESKTOP|KDE-i3|
|XDG_SESSION_TYPE|x11|
</details>

**Comment by erictraut** [comment]
Thanks for the bug report. This is a duplicate of https://github.com/microsoft/pyright/issues/3488, which was fixed in the latest version of pyright. Pylance will pick up the fix in the next release.

**Comment by debonte** [comment]
The fix for this issue is included in version 2022.5.3, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202253-25-May-2022)
